### PR TITLE
Set slip compliance

### PR DESCRIPTION
**Pull request by JShep1**
Required for the wheel slip plugin

### CODE REVIEW

File: dartsim/src/ShapeFeatures.hh [hunk 1]
```diff
@@ -150,6 +165,19 @@ class ShapeFeatures :
       const std::string &_name,
       const LinearVector3d &_normal,
       const LinearVector3d &_point) override;
+
```
Comment (azeey) [inline dartsim/src/ShapeFeatures.hh:168]: I think these are duplicated

File: include/ignition/physics/detail/Shape.hh [hunk 2]
```diff
@@ -99,6 +99,46 @@ namespace ignition
...
           ->SetShapeRestitutionCoefficient(this->identity, _other, _value);
     }
+
+    /////////////////////////////////////////////////
+    template <typename PolicyT, typename FeaturesT>
+    auto GetShapeFrictionPyramidSlipCompliance::Shape<PolicyT, FeaturesT>
+    ::GetPrimarySlipCompliance() const
+    -> Scalar
+    {
+      return this->template Interface<SetShapeFrictionPyramidSlipCompliance>()
```
Comment (azeey) [inline include/ignition/physics/detail/Shape.hh:109]: Set -> Get
```suggestion
      return this->template Interface<GetShapeFrictionPyramidSlipCompliance>()
```

File: include/ignition/physics/detail/Shape.hh [hunk 3]
```diff
@@ -99,6 +99,46 @@ namespace ignition
...
+          ->GetShapeFrictionPyramidPrimarySlipCompliance(this->identity);
+    }
+
+    /////////////////////////////////////////////////
+    template <typename PolicyT, typename FeaturesT>
+    auto GetShapeFrictionPyramidSlipCompliance::Shape<PolicyT, FeaturesT>
+    ::GetSecondarySlipCompliance() const
+    -> Scalar
+    {
+      return this->template Interface<SetShapeFrictionPyramidSlipCompliance>()
```
Comment (azeey) [inline include/ignition/physics/detail/Shape.hh:119]: Set -> Get
```suggestion
      return this->template Interface<GetShapeFrictionPyramidSlipCompliance>()
```

**Comment by scpeters** [comment]
@osrf-jenkins run tests please

**Comment by scpeters** [comment]
@osrf-jenkins run tests please

**Comment by scpeters** [comment]
@osrf-jenkins run tests please

**Comment by azeey** [comment]
@scpeters pointed me to another relevant test in Gazebo11 that would be nice to add here https://github.com/osrf/gazebo/blob/a9f093f2fc7fa7fc49efa73719f6536ad28f8af7/test/integration/physics_friction.cc#L529. The motion of the objects seem different from Gazebo-classic, so I'm investigating why. I think there is a bug in DART when the center of mass is offset from the origin of a link.

File: dartsim/src/ShapeFeatures.cc [hunk 4]
```diff
@@ -328,6 +328,80 @@ AlignedBox3d ShapeFeatures::GetShapeAxisAlignedBoundingBox(
   return AlignedBox3d(box.getMin(), box.getMax());
 }
 
+/////////////////////////////////////////////////
```
Comment (azeey) [inline dartsim/src/ShapeFeatures.cc:331]: Maybe we should guard all these features with `DART_VERSION_AT_LEAST(6, 10, 0)`. What do you think @scpeters ?

Comment (scpeters) [reply dartsim/src/ShapeFeatures.cc:331]: this is targeted at `ign-physics1`, which requires dart 6.10. I think we will need those guard conditions when we merge forward to `ign-physics2`.

**Comment by JShep1** [comment]
The wheel slip changes were merged into upstream dart, we will need new debs for CI.  All tests both on physics and gazebo are passing for me locally.  I'm holding off on merging these two PRs until we can get CI green.

**Comment by nkoenig** [comment]
DART has been released. I think this PR is next in line.

**Comment by nkoenig** [comment]
@osrf-jenkins run tests please

**Comment by azeey** [comment]
I was waiting on https://github.com/osrf/homebrew-simulation/pull/1130

**Comment by azeey** [comment]
@osrf-jenkins run tests please

**Comment by azeey** [comment]
@JShep1 Can you rerun github action CI? Seems like I don't have permission to do it.

**Comment by nkoenig** [comment]
> @JShep1 Can you rerun github action CI? Seems like I don't have permission to do it.

I don't understand why there is no button to re-run jobs. I think we all should have the same permissions.

**Comment by JShep1** [comment]
> @JShep1 Can you rerun github action CI? Seems like I don't have permission to do it.

Seems like they were triggered by your comment @azeey , they show as being started an hour ago when you made the comment, perhaps it took a little bit to trigger them?

**Comment by azeey** [comment]
> Seems like they were triggered by your comment @azeey , they show as being started an hour ago when you made the comment, perhaps it took a little bit to trigger them?

I meant just the [Github Actions CI](https://github.com/ignitionrobotics/ign-physics/pull/56/checks?check_run_id=956033967). My comment restarted the Jenkins CI, but not the Github Actions CI.

I also tried using the REST API to rerun it and it responded with "Unable to re-run this workflow run because it was created over a month ago". I think our best bet is to push an empty commit, but that will probably restart jenkins as well :(

**Comment by nkoenig** [comment]
> I also tried using the REST API to rerun it and it responded with "Unable to re-run this workflow run because it was created over a month ago". I think our best bet is to push an empty commit, but that will probably restart jenkins as well :(

And now the button on the github actions is there. So, an old PR can loose CI buttons. Interesting.

**Comment by chapulina** [comment]
:policewoman: 

There are a couple of old commits from @scpeters that aren't signed. I think it's fine to manually accept DCO for the PR, but please whoever squash-merges this PR, be sure to add his signature so we don't have issues merging this PR forward to other branches.

:arrow_right: `Signed-off-by: Steve Peters <scpeters@openrobotics.org>`

And always remember not to remove other signatures.

**Comment by nkoenig** [comment]
The Ubuntu CI keeps failing. The logs show:

```
2020-09-16T23:34:51.5187965Z ##[error]The operation was canceled.
```

during compilation. I'll try again.

**Comment by nkoenig** [comment]
> The Ubuntu CI keeps failing. The logs show:
> 
> ```
> 2020-09-16T23:34:51.5187965Z ##[error]The operation was canceled.
> ```
> 
> during compilation. I'll try again.

This has happened three times. Unless anyone has ideas, I'll merge this sometime tonight.

**Comment by azeey** [comment]
It could be timing out. ign-physics1 takes a really long time to compile on Ubuntu with tests enabled.

**Comment by nkoenig** [comment]
> It could be timing out. ign-physics1 takes a really long time to compile on Ubuntu with tests enabled.

It errors out at around 15-16 minutes. I'll try to muck with actions.

**Comment by chapulina** [comment]
wow the actions CI usually runs for close to 3 hours for `ign-physics1` :fearful: 

https://github.com/ignitionrobotics/ign-physics/actions?query=workflow%3A%22Ubuntu+CI%22+branch%3Aign-physics1

It's been over 2h 55min and it's still running for this branch

**Comment by nkoenig** [comment]
wow, it finished. I guess I was too impatient.

**Comment by nkoenig** [comment]
Okay. I think everything is fine. I'm merging.

**Comment by chapulina** [comment]
Please remember to squash-merge feature PRs. The commits from this PR are now interleaved with the rest of the commits in the release branches are are difficult to identify.

We don't completely lose the separate commits when squash-merging. The squashed commit has a link to the PR and the PR page displays all the individual commits.

**Pull request merged**